src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
>loweredRight = _factory.Convert(loweredLeft.Type, loweredRight); [](start = 24, length = 64)

The tests only cover conversion to object. Are there other conversions that would be interesting to test? For example, between interfaces and structures (both ways), between base and derived types (both ways), etc.